src/benchmarks/micro/runtime/Interop/ComWrappers.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Interop
+{
+    [BenchmarkCategory(Categories.Runtime)]
+    public unsafe class ComWrappersTests
+    {
+        static readonly ComWrappers s_instance = new DummyComWrappers();
+
+        static object QueryRCW(nint ptr)
+            => s_instance.GetOrCreateObjectForComInstance(ptr, CreateObjectFlags.None);
+
+        private List<nint> iterations;
+        private object targetRcw;
+
+        public ComWrappersTests()
+        {
+            // Allocate a simple IUnknown instance.
+            var ptr = (void**)NativeMemory.Alloc(5, (nuint)sizeof(void*));
+
+            ptr[1] = (void*)1; // Initial ref count.
+
+            // Define IUnknown vtable.
+            ptr[2] = (delegate* unmanaged<void*, Guid*, void**, int>)&QueryInterface;
+            ptr[3] = (delegate* unmanaged<void*, int>)&AddRef;
+            ptr[4] = (delegate* unmanaged<void*, int>)&Release;
+
+            // Assign vtable to instance.
+            ptr[0] = &ptr[2];
+
+            // Record RCW pointer.
+            nint rcwPtr = (nint)ptr;
+
+            // Transfer ownership to RCW instance. This populates the RCW cache.
+            targetRcw = s_instance.GetOrCreateObjectForComInstance((nint)ptr, CreateObjectFlags.None);
+
+            // Ownership was transferred.
+            Marshal.Release((nint)ptr);
+
+            // Define a large number of iterations for parallel action.
+            var limit = 3_000_000;
+            iterations = Enumerable.Repeat(rcwPtr, limit).ToList();
+        }
+
+        [Benchmark]
+        public unsafe ParallelLoopResult ParallelRCWLookUp()
+            => Parallel.ForEach(iterations, static (nint ptr) => QueryRCW(ptr));
+
+        sealed class DummyComWrappers : ComWrappers
+        {
+            protected override unsafe ComInterfaceEntry* ComputeVtables(object obj, CreateComInterfaceFlags flags, out int count)
+                => throw new NotImplementedException();
+
+            protected override object CreateObject(nint externalComObject, CreateObjectFlags flags)
+            {
+                Marshal.AddRef(externalComObject);
+                return new object();
+            }
+
+            protected override void ReleaseObjects(IEnumerable objects)
+                => throw new NotImplementedException();
+        }
+
+        static Guid IID_IUnknown = new Guid(0x00000000, 0x0000, 0x0000, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
```

Comment:
I'm surprised there isn't a convenient place this is already exposed?
Maybe
https://apisof.net/catalog/37ae96ea-3761-766e-fc9d-0d46cba60387

Reply:
There isn't a lot of value in consuming this from a NuGet package. It is a well defined constant where duplicates don't breed confusion in practice and pointing at a non-BCL library seems unnecessary. I do agree though that there is an argument to be made to add it to `Marshal` or perhaps the `ComWrappers` API.

Reply:
I've filed https://github.com/dotnet/runtime/issues/91150 to see if we can find a good place for some other COM IIDs.